tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java

```diff
@@ -0,0 +1,285 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.gradle.ui.quickfix;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.workspace.NewProjectHandler;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.compiler.IProblem;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ImportDeclaration;
+import org.eclipse.jdt.core.dom.Name;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ui.text.java.IInvocationContext;
+import org.eclipse.jdt.ui.text.java.IJavaCompletionProposal;
+import org.eclipse.jdt.ui.text.java.IProblemLocation;
+import org.eclipse.jdt.ui.text.java.IQuickFixProcessor;
+import org.eclipse.jdt.ui.text.java.correction.CUCorrectionProposal;
+import org.eclipse.jface.text.IDocument;
+
+import com.liferay.ide.gradle.core.GradleCore;
+import com.liferay.ide.gradle.core.parser.GradleDependency;
+import com.liferay.ide.gradle.core.parser.GradleDependencyUpdater;
+import com.liferay.ide.project.core.modules.ServiceContainer;
+import com.liferay.ide.project.core.util.TargetPlatformUtil;
+
+/**
+ * @author Lovett Li
+ */
+public class QuickFixGradleDep implements IQuickFixProcessor
+{
+
+    private IFile gradleFile;
+
+    @Override
+    public boolean hasCorrections( ICompilationUnit unit, int problemId )
+    {
+        switch( problemId )
+        {
+        case IProblem.ImportNotFound:
+        case IProblem.UndefinedType:
+        case IProblem.JavadocUndefinedType:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    @Override
+    public IJavaCompletionProposal[] getCorrections( IInvocationContext context, IProblemLocation[] locations )
+    {
+        if( locations == null || locations.length == 0 )
+        {
+            return null;
+        }
+
+        IProject project = context.getCompilationUnit().getResource().getProject();
+        gradleFile = project.getFile( "build.gradle" );
+        HashSet<Integer> handledProblems = new HashSet<>( locations.length );
```

Comment:
Don't put "hashSet" type on left, just use "Set<Integer>".  Also just use "new HashSet<>();" no need for default size.